src/utils/ProfileUtils.ts

```diff
@@ -56,6 +58,22 @@ async function extractConfigsToImportedProfile(
     }
 }
 
+export async function installModsAndResolveConflicts(
+    downloadedMods: ThunderstoreCombo[],
+    profile: Profile,
+    store: Store<any>
+): Promise<void> {
+    await ProfileModList.requestLock(async () => {
+        try {
+            const modList: ManifestV2[] = await installModsToProfile(downloadedMods, profile.asImmutableProfile());
+            await store.dispatch('profile/updateModList', modList);
+            throwForR2Error(await ConflictManagementProvider.instance.resolveConflicts(modList, profile));
```

Comment:
Downside of having a long running the stack of branches is that you'll need to rebase the whole stack to avoid conflicts. E.g. here your earlier changes mean this method no longer accepts Profile but ImmutableProfile, meaning `installModsAndResolveConflicts` should only deal in ImmutableProfiles too.

I guess we should try to merge the approved branches after the holidays even if the refactoring as whole isn't completed yet. In the meanwhile you might just keep doing the rebases manually, or try Graphite*, or do this shuffle where you keep only the latest approved PR (which contains the commits from earlier PRs too) and drop the earlier PRs so you'll only have one approved PR the rebase (this explanation probably sucked, we can go through what I meant in the daily).

*: I've personally used Graphite to handle the branches locally on my dev environment, put handled the pushing with vanilla git and created the PRs manually, as I'm not sure if it's good idea to include Graphite stuff in a repo where most other authors don't use it.

Reply:
We decided to close the earlier PRs and point this PR to develop, so it now contains already approved changes from earlier PRs.